libraries/api-cat-converters/src/main/java/no/fdk/acat/converters/apispecificationparser/OpenApiV3JsonParser.java

```diff
@@ -22,7 +22,29 @@ public boolean canParse(String spec) {
         try {
             JsonElement element = new JsonParser().parse(spec);
             String version = element.getAsJsonObject().get("openapi").getAsString();
-            return version.length() > 2 && version.substring(0, 2).equals("3.");
+            if (!(version.length() > 2 && version.substring(0, 2).equals("3."))) {
```

Comment:
Dette er inputfilparser. I teori har vi flere støttete formater. Det skalerer ikke hvis du sjekker det i hver eneste parser, det er vanskelig å holde de i synk.
Så jeg anbefaler å gjøre validering etter parse.
Det mener at klassestruktur er egentlig feil - jeg defineret interface så at alle parsere kunne tilsvare samme interface - universal er smart og gjør valgen selv, mens man kan bruke også spesifikk parser.

Et alternativ kunne også være at validering er ikke gjort som en del av parse, men utside etterpå.
Hvem kaller parse? De gjør vi i to plass:
api-cat konversjontjeneste (registration bruker dette)
fdk/applications/api-cat/src/main/java/no/acat/controller/ConvertController.java:43

final konversion før lagring as søkindeks
fdk/applications/api-cat/src/main/java/no/acat/service/ApiDocumentBuilderService.java:49

Vil ikke, rote client-kode, derfor "konversion"-tjeneste må gi ut full-vlidert svar.
1) I Parser interface skal ha nytt metod ParseAndValidate. Men den løsning har fortsett samme problem, at hver implementasjonklass (vi har tre implementasjoner!) må implementere det.

2) la parser interface være som de er, men ha nytt konseptsion til validerte.
UniversalParser skulle ble re-navnt til ApiSpecConverter. Den følger Ikke Parser interface igjen. Den eneste metod den eksponerer er ApiSpecification ParseAndValidate(String spec) 

Så egentlig, 2 er mye bedre valg en 1)

Reply:
Det gir ikke mening å sjekke dette etter parse. Det er når vi først oppdager dokumentet at vi vil si om det er ok eller ikke. Det er et gjennomgående prinsipp at vi vil gjemme unna så mye kompleksitet som mulig i randsonen av applikasjonen.